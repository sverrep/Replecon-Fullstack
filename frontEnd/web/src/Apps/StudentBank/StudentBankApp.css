.bank-info{
    margin: auto;
    width: 60%;
    margin-top: 15px;
    padding: 10px;
    text-align: center;
}

.wrapper{
    font-family: 'Montserrat', sans-serif;
}

.con{
    margin-left: 40px;
    margin-top: 20px;
}

.bank-cards{
    margin-top: 20px;
}