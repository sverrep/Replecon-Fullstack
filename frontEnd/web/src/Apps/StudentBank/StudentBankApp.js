import React from 'react';
import { withRouter } from "react-router-dom";
import axios from 'axios';
import getIP from '../../settings.js';
import Navbar from '../../Components/navbar/Student NavBar/Navbar.js';
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import Card from 'react-bootstrap/Card';
import Row from 'react-bootstrap/Row';
import Alert from 'react-bootstrap/Alert'
import './StudentBankApp.css';
import Cookies from 'universal-cookie';

class StudentBank extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            bank_name: '',
            class_name: '',
            classroom: '',
            interest_rate:'',
            payout_rate:'',
            student_balance: '',
            show:false,
            value:'',
            loggedin_student:{},
            savings: [],

            showAlert: false,
            variant: '',
            message: '',



        }
        this.getClassroomDetails = this.getClassroomDetails.bind(this)
        this.findClassroom = this.findClassroom.bind(this)
        this.getClassStudents = this.getClassStudents.bind(this)
        this.getBanks = this.getBanks.bind(this)
        this.getStudentBalance = this.getStudentBalance.bind(this)
    }

    async componentDidMount(){
      const cookies = new Cookies()
      axios.defaults.headers.common.Authorization = cookies.get("Authorization");
      await this.getClassStudents()
      await this.getBanks()
      await this.getStudentBalance()

    }

    async getStudentBalance(){
      await axios.get(getIP()+'/students/balance/')
      .then(response => {
        this.setState({student_balance: response.data})
      })
      .catch(error => this.setState({message: error}))
      }
    //Retrieving the class name
    async getClassStudents() {
        await axios.get(getIP()+'/students/class_code/')
        .then(async response  => {
            this.setState({ classroom: response.data[0].class_code })
            this.getBanks(response.data[0].class_code)
            this.getClassroomDetails()
            await axios.get(getIP()+'/students/current/')
              .then(response => {
                this.setState({loggedin_student: response.data}, () => {this.getStudentSavings()})
              })
        })
        .catch(error => console.log(error))
    }

    async getClassroomDetails(){
        await axios.get(getIP()+'/classrooms/')
        .then(response => {
            this.findClassroom(response.data)
        })
        .catch(error => console.log(error))
    }
    
    findClassroom(classrooms){
        for (let i = 0; i <= classrooms.length-1; i++)
        {
          if (classrooms[i].class_code === this.state.classroom)
          {
            this.setState({ class_name: classrooms[i].class_name });
          }
        }
    }
    
    //Getting the inrest rate

    async getBanks(class_code){
        await axios.get(getIP()+'/banks/')
        .then(response => {
          this.setState({banks: response.data})
          this.getBankDetails(response.data, class_code)
        })
        .catch(error => console.log(error))
      }
    
      getBankDetails(banks, class_code){
        for (let i = 0; i<=banks.length -1;i++)
        {
          if(banks[i].class_code === class_code)
          {
            this.setState({interest_rate:banks[i].interest_rate})
            this.setState({payout_rate:banks[i].payout_rate})
          }
        }
      }

      //Insert Savings
      async setStudentSavings(){
        if(await this.validateSavings()){
          await axios.post(getIP()+'/transactions/banksavings/', {"amount": this.state.value, "done": false})
          .then(async response => {
            var transaction_id = response.data["id"]
            await axios.post(getIP()+'/transactioninterestrates/', {"class_code": this.state.classroom, "transaction_id": transaction_id})
            .then(async response => {
              await axios.get(getIP()+'/students/bank/')
              .then(async response => {
                await axios.put(getIP()+'/students/balance/', { amount: this.state.value, user_id: response.data, recipient: false })
                .then(response => {
                  this.setState({show:false})
                  this.getStudentSavings()
                  this.getStudentBalance()
                })
                .catch(error => console.log(error))
              })
              .catch(error => console.log(error))
            })
            .catch(error => console.log(error))
          })
          .catch(error => console.log(error))
        }
      }
      //Start Saving Modal
      openModal(){
        this.setState({show:true})
      }

      closeModal(){
        this.setState({show:false})
      }



      renderModal(){
          return(
            <Modal
            show={this.state.show}
            backdrop="static"
            keyboard={false}
            >
                <Modal.Header>
                    <Modal.Title>Start Saving</Modal.Title>
                </Modal.Header>

                <Modal.Body>
                    <Form>
                    <Form.Group className="mb-3" controlId="formBasicEmail">
                        
                        <Form.Control onChange={e => this.setState({value: e.target.value})} type="number" placeholder="Enter the amount to save" />
                    </Form.Group>
                    </Form>
                </Modal.Body>

                <Modal.Footer>
                    <Button variant="secondary" onClick={() => this.closeModal()}>Cancel</Button>
                    <Button variant="primary" className='bank-button' onClick={() => this.setStudentSavings()}>Confirm Saving</Button>
                </Modal.Footer>
            </Modal>
        )
      }


      //Get Savings
      async getStudentSavings(){
        this.setState({savings: []})
        await axios.get(getIP()+'/transactioninterestrates/')
        .then(async response1 => {
          
            for(let i = 0; i <= response1.data.length-1; i++)
            {
                await axios.get(getIP()+'/transactions/' + response1.data[i].transaction_id)
                .then(async response2 => {
                    var initamount = parseFloat(response2.data.amount)
                    if(this.state.loggedin_student.id === response2.data.sender_id)
                    {
                        var intrate = parseFloat(response1.data[i].set_interest_rate)
                        var finalamount = initamount + (initamount*(intrate/100))
                        await axios.get(getIP()+'/transactioninterestrates/payoutdate/' + response2.data.id)
                        .then(response => {
                          var payout_date = (((response.data / 60) / 60) / 24)
                          var tempdict = {
                            "id": i, 
                            "initial_amount": initamount, 
                            "interest_rate": intrate, 
                            "final_amount": finalamount, 
                            "transaction_id": response2.data.id, 
                            "payout_date": payout_date, 
                            "active": response1.data[i].active
                          }
                          this.setState({message: tempdict})
                          this.setState({savings: [...this.state.savings, tempdict]})
                        })
                        .catch(error => console.log(error))
                    }
                })
                .catch(error => console.log(error))
            }
        })
        .catch(error => console.log(error))
      }

      renderSavings(item, i){
        if(item.active === true)
    {
        return(
          <Card key={i} style={{ width: '15rem' }} className='cards'>
              <Card.Body>
                  <Card.Title>Amount: {item.initial_amount}</Card.Title>
                  <Card.Text>
                      Payout of {item.final_amount}$ in {item.payout_date} days
                  </Card.Text>
                  <Button variant="primary" className='bank-button' onClick={() => this.claimSavings(item)}>Claim Savings</Button>
              </Card.Body>
          </Card>
      )
      }
    }

    async validateClaim(item){
      var temp = false
      await this.getStudentSavings()
      for(let i = 0; i<= this.state.savings.length -1; i++){
        if(this.state.savings[i].id===item.id){
          if(this.state.savings[i].active)
          temp = true
        }
      }
      if(temp === true){
        if(item.payout_date <= 0){
          this.setState({variant:'success'})
          this.setState({message:'Money claimed successfully'})
          this.setState({showAlert:true})
          return true
        }
        else{
          this.setState({variant:'danger'})
          this.setState({message:'That money is not ready to be claimed'})
          this.setState({showAlert:true})
          return false
        }
      }
      else{
        console.log('you can not do that :)))))')
        return false
        
      }
    }
        async claimSavings(item){
          
          if(await this.validateClaim(item))
          {
            await axios.post(getIP()+'/transactions/banksavings/', {"amount": Math.round(item.final_amount * 10) / 10, "done": true})
            .then(async response => {
              await axios.put(getIP()+'/transactioninterestrates/', {active: false, class_code: this.state.classroom, transaction_id: item.transaction_id})
              .then(async response => {
                await axios.get(getIP()+'/students/bank/')
                .then(async response => {
                  await axios.put(getIP()+'/students/balance/', { amount: (Math.round(item.final_amount * 10) / 10).toString(), user_id: response.data, recipient: true })
                  .then(response => {
                    this.getStudentSavings()
                    this.getStudentBalance()
                  })
                  .catch(error => console.log(error))
                })
                .catch(error => console.log(error))
              })
              .catch(error => console.log(error))
            })
            .catch(error => console.log(error))
          }
        }

    
    async validateSavings(){
      await this.getStudentBalance()
        if(isNaN(this.state.value)){
          this.setState({variant:'danger'})
          this.setState({message:'Please enter a vaild number'})
          this.setState({showAlert:true})
          this.closeModal()
          return false
        }
        else if(Math.sign(this.state.value)<0 || this.state.value === '0'){
            this.setState({variant:'danger'})
            this.setState({message:'Please enter a postive number'})
            this.setState({showAlert:true})
            this.closeModal()
            return false
        }
        else{
            if(this.state.value>this.state.student_balance){
              this.setState({variant:'danger'})
              this.setState({message:'You dont have that amount of money'})
              this.setState({showAlert:true})
              this.closeModal()
              return false
            }
            else{
              this.setState({variant:'success'})
              this.setState({message:'Money saved successfully'})
              this.setState({showAlert:true})
              return true
            }
        }
    }
    renderAlert(variant, message){
      if(this.state.showAlert){
        return(
        <Alert show={this.state.showAlert} variant={variant} onClose={() => this.setState({showAlert:false})} dismissible>
            <p>
                {message}
            </p>
        </Alert>
        )
        }
    }
        
    render(){
        return(
        <div className='wrapper'>
            <Navbar/>
            <div className='bank-content'>
            <div>
                <div className='bank-title'>
                  <div className='title-bank'><h3>Bank of {this.state.class_name}</h3></div>
                  <div className='title-balance'><h3>${this.state.student_balance}</h3></div>
                </div>

                <div className='bank-title'>
                  <div className='title-bank'><h6>Current Intrest Rate: {this.state.interest_rate} %</h6>
                <h6>Current Payout Rate: {this.state.payout_rate} week</h6>
                <h6>Current Balance: {this.state.student_balance}$</h6></div>
                  <div className='title-balance'><h3><Button className='bank-button' onClick={() => this.openModal()}>
                  Start Saving
                </Button></h3></div>
                </div>
                
                
                
                
                {this.renderModal()}
            </div>
            <div className='alert'>
                {this.renderAlert(this.state.variant, this.state.message)}
            </div>
            <div className='con'>
                <h2>My Savings</h2>
                <div className="savings_cards">
                    <div className='bank-cards'>
                        <Row xs="auto" md={2} className='override'>
                          {this.state.savings.map((item, i) => this.renderSavings(item, i))}
                        
                        </Row>
                      </div>
                </div>
            </div>
            </div>
        </div>
        )
    }

}

export default withRouter(StudentBank)