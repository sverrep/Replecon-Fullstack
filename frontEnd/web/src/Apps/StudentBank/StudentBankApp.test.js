import React from 'react';
import StudentBankApp from './StudentBankApp.js'
import axios from 'axios'

jest.mock('axios');
let wrapper
const props = {
    location:{
        state:{
            role: "Student",
            token: 'Token'
        }
    }  
}

describe('Student Bank Tests', ()=>{
    
    beforeEach(() => {
        wrapper = shallow(<StudentBankApp.WrappedComponent {...props}/>, { disableLifecycleMethods: true })
    });

    describe('Component Did Mount', ()=>{
        it('should call correct functions in component did mount', async()=>{
            const instance = wrapper.instance()
            instance.getClassStudents = jest.fn()
            instance.getClassStudents.mockReturnValueOnce()
    
            instance.getBanks = jest.fn()
            instance.getBanks.mockReturnValueOnce()
    
            instance.getStudentBalance = jest.fn()
            instance.getStudentBalance.mockReturnValueOnce()
    
            await instance.componentDidMount()
    
            expect(instance.getClassStudents).toHaveBeenCalledTimes(1)
            expect(instance.getBanks).toHaveBeenCalledTimes(1)
            expect(instance.getStudentBalance).toHaveBeenCalledTimes(1)
        })
    })

    describe('Student Balance', ()=>{
        it('should retrieve the proper student balance', async()=>{
            const instance = wrapper.instance()
            const bal = mockResponseData('100')
            axios.get.mockImplementationOnce(() =>  Promise.resolve(bal))
            await instance.getStudentBalance()
            
            expect(axios.get).toHaveBeenCalledTimes(1)
            expect(instance.state.student_balance).toBe('100')
        })

        it('should not retrieve the student balance', async()=>{
            const instance = wrapper.instance()
            axios.get.mockImplementationOnce(() =>  Promise.reject('Token is not authenticated'))
            await instance.getStudentBalance()
            
            expect(axios.get).toHaveBeenCalledTimes(1)
            expect(instance.state.message).toBe('Token is not authenticated')
        })
    })

    describe('Class Students', ()=>{
        it('should get student classroom', async()=>{
            const instance = wrapper.instance()

            instance.getBanks = jest.fn()
            instance.getBanks.mockReturnValueOnce()
    
            instance.getClassroomDetails = jest.fn()
            instance.getClassroomDetails.mockReturnValueOnce()
    
            instance.getStudentSavings = jest.fn()
            instance.getStudentSavings.mockReturnValueOnce()
    
            const classroom = mockResponseData([{class_code: '2'}, {class_code: '2'}])
            axios.get.mockImplementationOnce(() =>  Promise.resolve(classroom))
    
            const user = mockResponseData({class_code: '2', name: 'name', balance: '123'})
            axios.get.mockImplementationOnce(() =>  Promise.resolve(user))
    
            await instance.getClassStudents()
            expect(axios.get).toHaveBeenCalledTimes(2)
            expect(instance.state.classroom).toBe('2')
            expect(instance.state.loggedin_student).toStrictEqual({class_code: '2', name: 'name', balance: '123'})
        })

        it('should get classroom details', async()=>{
            const instance = wrapper.instance()
            const classroom = mockResponseData([{classroom: 'classroom of 1'}, {classroom: 'classroom of 2'}])
            axios.get.mockImplementationOnce(() =>  Promise.resolve(classroom))

            instance.findClassroom = jest.fn()
            instance.findClassroom.mockReturnValueOnce()

            await instance.getClassroomDetails()

            expect(instance.findClassroom).toHaveBeenCalledTimes(1)
            expect(axios.get).toHaveBeenCalledTimes(1)
        })

        it('should find a classroom', ()=>{
            const instance = wrapper.instance()
            instance.state.classroom = '123'
            const classrooms = [{class_code: '123', class_name: 'best classroom'}, {class_code: '452', class_name: 'worst classroom'}]
            instance.findClassroom(classrooms)
            expect(instance.state.class_name).toBe('best classroom')
        })

        it('should not find a classroom', ()=>{
            const instance = wrapper.instance()
            instance.state.classroom = '555'
            const classrooms = [{class_code: '123', class_name: 'best classroom'}, {class_code: '452', class_name: 'worst classroom'}]
            instance.findClassroom(classrooms)
            expect(instance.state.class_name).toBe('')
        })
        

    })

    describe('Class Banks', ()=>{
        it('should get banks', async()=>{
            const instance = wrapper.instance()
            
            const banks = mockResponseData('banks details')
            axios.get.mockImplementationOnce(() =>  Promise.resolve(banks))
            
            instance.getBankDetails = jest.fn()
            instance.getBankDetails.mockReturnValueOnce()
            await instance.getBanks('123')
            expect(instance.state.banks).toBe('banks details')
            expect(instance.getBankDetails).toHaveBeenCalledTimes(1)
        })

        it('should find the bank details', ()=>{
            const instance = wrapper.instance()
            const banks = [{class_code: '123', interest_rate: '12', payout_rate: '1'}, {class_code: '231', interest_rate: '15', payout_rate: '2'}]
            const class_code = '123'
            instance.getBankDetails(banks, class_code)
            expect(instance.state.interest_rate).toBe('12')
            expect(instance.state.payout_rate).toBe('1')
        })

        it('should not find a bank detail', ()=>{
            const instance = wrapper.instance()
            const banks = [{class_code: '123', interest_rate: '12', payout_rate: '1'}, {class_code: '231', interest_rate: '15', payout_rate: '2'}]
            const class_code = '555'
            instance.getBankDetails(banks, class_code)
            expect(instance.state.interest_rate).toBe('')
            expect(instance.state.payout_rate).toBe('')
        })
    })

    describe('Student Savings', ()=>{
        it('should find the student savings get', async()=>{
            const instance = wrapper.instance()
            instance.state.loggedin_student = {id: '2'}
            
            const transactions = mockResponseData([
            {transaction_id: '1', active: true, set_interest_rate: '10'}, 
            {transaction_id: '2', active: true, set_interest_rate: '15'}
            ])
            axios.get.mockImplementationOnce(() =>  Promise.resolve(transactions))

            const firsttrans = mockResponseData({amount: '100', sender_id: '2', id: '213'})
            axios.get.mockImplementationOnce(() =>  Promise.resolve(firsttrans))

            const firstpayout = mockResponseData('12/12/2021')
            axios.get.mockImplementationOnce(() =>  Promise.resolve(firstpayout))

            const secondtrans = mockResponseData({amount: '100', sender_id: '1', id: '215'})
            axios.get.mockImplementationOnce(() =>  Promise.resolve(secondtrans))

            await instance.getStudentSavings()
            expect(instance.state.savings.length).toStrictEqual(1)
            expect(axios.get).toHaveBeenCalledTimes(4)
            
            const expected_result = [{
                "active": true,
                 "final_amount": 110, 
                 "id": 0, "initial_amount": 100, 
                 "interest_rate": 10, 
                 "payout_date": NaN, 
                 "transaction_id": "213"}]

            expect(instance.state.savings).toStrictEqual(expected_result)
        })

        it('should not find any savings get', async()=>{
            const instance = wrapper.instance()
            instance.state.loggedin_student = {id: '2'}
            
            const transactions = mockResponseData([
            {transaction_id: '1', active: true, set_interest_rate: '10'}, 
            {transaction_id: '2', active: true, set_interest_rate: '15'}
            ])
            axios.get.mockImplementationOnce(() =>  Promise.resolve(transactions))

            const firsttrans = mockResponseData({amount: '100', sender_id: '3', id: '213'})
            axios.get.mockImplementationOnce(() =>  Promise.resolve(firsttrans))
            const secondtrans = mockResponseData({amount: '100', sender_id: '1', id: '215'})
            axios.get.mockImplementationOnce(() =>  Promise.resolve(secondtrans))

            await instance.getStudentSavings()
            expect(instance.state.savings.length).toStrictEqual(0)
            expect(axios.get).toHaveBeenCalledTimes(3)
            expect(instance.state.savings).toStrictEqual([])
        })

        it('should set the student savings', async()=>{
            const instance = wrapper.instance()
            instance.state.value = '100'
            instance.state.classroom = '123'

            const payload = {"amount": '100', "done": false}
            const banksavings = mockResponseData({id: '12'})
            axios.post.mockImplementationOnce(() => Promise.resolve(banksavings))

            const payload2 = {"class_code": '123', "transaction_id": '12'}
            
            axios.post.mockImplementationOnce(() => Promise.resolve())

            const bankid = mockResponseData('123')
            axios.get.mockImplementationOnce(() =>  Promise.resolve(bankid))

            const payload3 = { amount: '100', user_id: '123', recipient: false }
            axios.put.mockImplementationOnce(() =>  Promise.resolve())

            instance.getStudentSavings = jest.fn()
            instance.getStudentSavings.mockImplementationOnce()

            instance.validateSavings = jest.fn()
            instance.validateSavings.mockReturnValueOnce(true)

            instance.getStudentBalance = jest.fn()
            instance.getStudentBalance.mockImplementationOnce()



            await instance.setStudentSavings()
            expect(instance.state.show).toBe(false)
            expect(axios.post).toHaveBeenCalledTimes(2)
            expect(axios.put).toHaveBeenCalledTimes(1)
            expect(axios.get).toHaveBeenCalledTimes(1)
            expect(instance.getStudentSavings).toHaveBeenCalledTimes(1)
            expect(instance.getStudentBalance).toHaveBeenCalledTimes(1)
        })

        it('should claim the student savings', async()=>{
            const instance = wrapper.instance()
            const item ={final_amound: '100', transaction_id: '111'}
            

            
            axios.post.mockImplementationOnce(() => Promise.resolve())
            axios.put.mockImplementationOnce(() => Promise.resolve())

            const bankid = mockResponseData('123')
            axios.get.mockImplementationOnce(() =>  Promise.resolve(bankid))
            axios.put.mockImplementationOnce(() => Promise.resolve())

            instance.getStudentSavings = jest.fn()
            instance.getStudentSavings.mockImplementationOnce()

            instance.validateClaim = jest.fn()
            instance.validateClaim.mockReturnValueOnce(true)

            instance.getStudentBalance = jest.fn()
            instance.getStudentBalance.mockImplementationOnce()
            
            await instance.claimSavings(item)
    
            expect(axios.put).toHaveBeenCalledTimes(2)
            expect(axios.get).toHaveBeenCalledTimes(1)
            expect(axios.post).toHaveBeenCalledTimes(1)
            expect(instance.getStudentSavings).toHaveBeenCalledTimes(1)
            expect(instance.getStudentBalance).toHaveBeenCalledTimes(1)
        })
    })

    describe('Validation', ()=>{
        it("should pass claim validation", ()=>{
            const instance = wrapper.instance()
            const item = {payout_date: 0}
            const result = instance.validateClaim(item)
            expect(result).toBe(true)
            expect(instance.state.message).toBe('Money claimed successfully')
            expect(instance.state.showAlert).toBe(true)
        })

        it('should fail claim validation', ()=>{
            const instance = wrapper.instance()
            const item = {payout_date: 2}
            const result = instance.validateClaim(item)
            expect(result).toBe(false)
            expect(instance.state.message).toBe('That money is not ready to be claimed')
            expect(instance.state.showAlert).toBe(true)
        })

        it('should pass validate savings', ()=>{
            const instance = wrapper.instance()
            instance.state.value = '123'
            instance.state.student_balance = '200'
            const result = instance.validateSavings()

            expect(result).toBe(true)
            expect(instance.state.message).toBe('Money saved successfully')
        })

        it('should fail validate savings', ()=>{
            const instance = wrapper.instance()
            instance.state.value = '123'
            instance.state.student_balance = '100'
            const result = instance.validateSavings()

            expect(result).toBe(false)
            expect(instance.state.message).toBe('You dont have that amount of money')
        })
    })

    describe('Renders', ()=>{
        it('should render page properly', ()=>{
            expect(wrapper.getElements()).toMatchSnapshot()
        })

        it('should render card properly', ()=>{
            const instance = wrapper.instance()
            const item = {intial_amaount: '123', final_amount: '200', payout_date: '3', active: true}
            expect(instance.renderSavings(item, 1)).toMatchSnapshot()
        })

        it('should render alert properly', ()=>{
            const instance = wrapper.instance()
            expect(instance.renderAlert('danger', 'message')).toMatchSnapshot()
        })
    })
})


const mockResponseData = jest.fn((data) => {
    if (typeof data === 'string')
    {
        return {
            data,
        };
    }
    else
    {
        return {
            data,
        };
    }
  });