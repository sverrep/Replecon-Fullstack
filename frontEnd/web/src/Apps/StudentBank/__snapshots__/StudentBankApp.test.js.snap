// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Bank Tests Renders should render alert properly 1`] = `undefined`;

exports[`Student Bank Tests Renders should render card properly 1`] = `
<Card
  body={false}
  className="cards"
  style={
    Object {
      "width": "15rem",
    }
  }
>
  <CardBody>
    <CardTitle>
      Amount: 
    </CardTitle>
    <CardText>
      Payout of 
      200
      $ in 
      3
       days
    </CardText>
    <Button
      active={false}
      className="bank-button"
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Claim Savings
    </Button>
  </CardBody>
</Card>
`;

exports[`Student Bank Tests Renders should render page properly 1`] = `
Array [
  <div
    className="wrapper"
  >
    <Navbar
      className="teacher-navbar"
      collapseOnSelect={true}
      expand="lg"
      variant="dark"
    >
      <Container
        fluid={false}
      >
        <NavbarBrand
          className="navbar-header"
        >
          Replecon
        </NavbarBrand>
        <Nav
          className="navbar-btns"
          fill={false}
          justify={false}
        >
          <Button
            active={false}
            className="navbar-btn"
            disabled={false}
            variant="light"
          >
            <Link
              className="navbar-link"
              to={
                Object {
                  "pathname": "/Profile",
                  "state": Object {
                    "role": "Student",
                    "token": "Token",
                  },
                }
              }
            >
              Profile
            </Link>
          </Button>
          <Button
            active={false}
            className="navbar-btn"
            disabled={false}
            variant="light"
          >
            <Link
              className="navbar-link"
              to={
                Object {
                  "pathname": "/StudentStore",
                  "state": Object {
                    "role": "Student",
                    "token": "Token",
                  },
                }
              }
            >
              Store
            </Link>
          </Button>
          <Button
            active={false}
            className="navbar-btn"
            disabled={false}
            variant="light"
          >
            <Link
              className="navbar-link"
              to={
                Object {
                  "pathname": "/StudentBank",
                  "state": Object {
                    "role": "Student",
                    "token": "Token",
                  },
                }
              }
            >
              Bank
            </Link>
          </Button>
          <Button
            active={false}
            className="navbar-btn"
            disabled={false}
            variant="light"
          >
            <Link
              className="navbar-link"
              to={
                Object {
                  "pathname": "/StudentClass",
                  "state": Object {
                    "role": "Student",
                    "token": "Token",
                  },
                }
              }
            >
              Classroom
            </Link>
          </Button>
        </Nav>
      </Container>
    </Navbar>
    <div
      className="bank-content"
    >
      <div>
        <div
          className="bank-title"
        >
          <div
            className="title-bank"
          >
            <h3>
              Bank of 
              
            </h3>
          </div>
          <div
            className="title-balance"
          >
            <h3>
              $
              
            </h3>
          </div>
        </div>
        <div
          className="bank-title"
        >
          <div
            className="title-bank"
          >
            <h6>
              Current Intrest Rate: 
              
               %
            </h6>
            <h6>
              Current Payout Rate: 
              
               week
            </h6>
            <h6>
              Current Balance: 
              
              $
            </h6>
          </div>
          <div
            className="title-balance"
          >
            <h3>
              <Button
                active={false}
                className="bank-button"
                disabled={false}
                onClick={[Function]}
                variant="primary"
              >
                Start Saving
              </Button>
            </h3>
          </div>
        </div>
        <Modal
          animation={true}
          autoFocus={true}
          backdrop="static"
          dialogAs={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          enforceFocus={true}
          keyboard={false}
          restoreFocus={true}
          show={false}
        >
          <ModalHeader
            closeButton={false}
            closeLabel="Close"
          >
            <ModalTitle>
              Start Saving
            </ModalTitle>
          </ModalHeader>
          <ModalBody>
            <Form>
              <FormGroup
                className="mb-3"
                controlId="formBasicEmail"
              >
                <FormControl
                  onChange={[Function]}
                  placeholder="Enter the amount to save"
                  type="number"
                />
              </FormGroup>
            </Form>
          </ModalBody>
          <ModalFooter>
            <Button
              active={false}
              disabled={false}
              onClick={[Function]}
              variant="secondary"
            >
              Cancel
            </Button>
            <Button
              active={false}
              className="bank-button"
              disabled={false}
              onClick={[Function]}
              variant="primary"
            >
              Confirm Saving
            </Button>
          </ModalFooter>
        </Modal>
      </div>
      <div
        className="alert"
      />
      <div
        className="con"
      >
        <h2>
          My Savings
        </h2>
        <div
          className="savings_cards"
        >
          <div
            className="bank-cards"
          >
            <Row
              className="override"
              md={2}
              xs="auto"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
